test.js

```diff
@@ -1,147 +1,107 @@
-      ]);
-    }, 1000);
-  });
+// A non-optimized to-do list application with redundant logic and poor performance
+let toDoList = [];
+// Function to add a task (overcomplicated and inefficient)
+function addTask(task) {
+ if (task === "") {
+ alert("Task cannot be empty!");
+ return;
+ }
+ let taskId = Date.now(); // Using Date.now() as a unique ID, not optimal for large data
+ let taskObject = {
+ id: taskId,
+ text: task,
+ completed: false
+ };
+ toDoList.push(taskObject);
+ // Rebuild the entire list every time
+ let listContainer = document.getElementById('task-list');
+ listContainer.innerHTML = ''; // Inefficient clearing of the entire list
+ for (let i = 0; i < toDoList.length; i++) {
+ let taskItem = document.createElement('li');
+ taskItem.textContent = toDoList[i].text;
+ listContainer.appendChild(taskItem);
+ }
 }
-
-// DOM Elements (Missing element IDs, some will not work)
-const taskList = document.getElementById('task-list');
-const taskInput = document.getElementById('taskinput'); // Typo here: 'task-input' is the actual ID
-const addButton = document.getElementById('add-button');
-const errorMessage = document.getElementById('error-message');
-
-// Task storage (in-memory for now)
-let tasks = [];
-
-// Function to render tasks to the screen
-function renderTasks() {
-  taskList.innerHTML = ''; // Clear the list before rendering
-
-  if (tasks.length === 0) {
-    taskList.innerHTML = '<li>No tasks to show</li>';
-  } else {
-    tasks.forEach(task => {
-      const taskElement = document.createElement('li');
-      taskElement.className = task.completed ? 'completed' : '';
-
-      const taskText = document.createElement('span');
-      taskText.textContent = task.text;
-      taskText.className = 'task-text';
-      taskElement.appendChild(taskText);
-
-      // Edit Button
-      const editButton = document.createElement('button');
-      editButton.textContent = 'Edit';
-      editButton.className = 'edit-btn';
-      editButton.onclick = () => editTask(task.id);
-      taskElement.appendChild(editButton);
-
-      // Delete Button (Forgot to add a delete button handler)
-      const deleteButton = document.createElement('button');
-      deleteButton.textContent = 'Delete';
-      deleteButton.className = 'delete-btn';
-      taskElement.appendChild(deleteButton);
-
-      // Mark as Completed Button
-      const completeButton = document.createElement('button');
-      completeButton.textContent = task.completed ? 'Undo' : 'Complete';
-      completeButton.className = 'complete-btn';
-      completeButton.onclick = () => toggleComplete(task.id);
-      taskElement.appendChild(completeButton);
-
-      taskList.appendChild(taskElement);
-    });
-  }
+// Function to mark a task as completed (no proper search)
+function markTaskAsCompleted(taskId) {
+ for (let i = 0; i < toDoList.length; i++) {
+ if (toDoList[i].id === taskId) {
+ toDoList[i].completed = true;
+ break;
```

Comment:
Line 30: Code style suggestion: Ensure consistent indentation and spacing for better readability.